src/Agent/NewRelic/Profiler/Profiler/Function.h

```diff
@@ -335,6 +336,15 @@ namespace NewRelic { namespace Profiler
                 _tracerFlags |= NewRelic::Profiler::Configuration::TracerFlags::AsyncMethod;
             }
 
+            mdAssembly mda = 0;
+            _metaDataAssemblyImport->GetAssemblyFromScope(&mda);
```

Comment:
We should consider using the ThrowOnError helper method for this api call. At the very least we'll need to check the hresult by using the SUCCEEDED macro or comparing against S_OK.